source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -775,13 +776,25 @@ namespace
             for (InputPtr input : surfaceShader->getActiveInputs())
             {
                 BindInputPtr bindInput = shaderRef->addBindInput(input->getName(), input->getType());
-                if (input->hasNodeName())
+                if (input->hasNodeName() && input->hasOutputString())
                 {
-                    if (input->hasOutputString())
+                    if (doc->getNodeGraph(input->getNodeName()))
```

Comment:
Thanks for managing the lack of "nodename" support (one thing upgrading to a node tries to address).